linker/shared/src/main/scala/org/scalajs/linker/frontend/optimizer/GenIncOptimizer.scala

```diff
@@ -136,83 +128,61 @@ abstract class GenIncOptimizer private[optimizer] (config: CommonPhaseConfig) {
    *  UPDATE PASS ONLY. (This IS the update pass).
    */
   private def updateAndTagEverything(linkedClasses: List[LinkedClass]): Unit = {
+    val neededStaticLikes = CollOps.emptyParMap[String, LinkedClass]
     val neededClasses = CollOps.emptyParMap[String, LinkedClass]
-    val neededStatics = CollOps.emptyParMap[String, LinkedClass]
-    val neededDefaults = CollOps.emptyParMap[String, LinkedClass]
     for (linkedClass <- linkedClasses) {
       // Update the list of ancestors for all linked classes
       getInterface(linkedClass.encodedName).ancestors = linkedClass.ancestors
 
+      CollOps.put(neededStaticLikes, linkedClass.encodedName, linkedClass)
+
```

Comment:
Just to test my understanding: This is a simplification that we say every class needs it's own static namespace unconditionally. Since changing the static namespace tags the class for update anyways.